hugo/content/2-models/allowed-types.md

```diff
@@ -0,0 +1,173 @@
+            "address" : "247 E 26 St, New York, NY"
+        },
+        {
+            "type" : "work",
+            "phone" : "555-123-4567",
+            "address" : "570 Second Ave, New York, NY"
+        }
+    ]
+}
+```
+
+The only restriction regarding serializable subdocuments is that they must be *default constructible*.
+
+## Optionals
+
+Although {{% a_blank "`std::optional<T>`" "http://en.cppreference.com/w/cpp/utility/optional/optional" %}} won't be available until C++17, Mangrove offers it as a serializable type in the form of `bson_mapper::stdx::optional<T>`. `optional` is a powerful new C++ type that specifies a value that may or may not exist.
+
+Mangrove offers support for optional types where `T` is a type discussed on this page. This gives Mangrove support for dynamic, flexible schemas, which will be discussed in a [later section](/2-models/dynamic-schemas).
+
+## ObjectIDs
+
+One of the fundamental types in MongoDB is the {{% a_blank "ObjectID" "https://docs.mongodb.com/manual/reference/bson-types/#objectid" %}}. ObjectIDs are the default type used for the `_id` field in documents, so you may want to use them in your project.
+
+The MongoDB C++ Driver offers the {{% a_blank "`bsoncxx::oid`" "http://mongodb.github.io/mongo-cxx-driver/classbsoncxx_1_1oid.html" %}}, which is a simple class that can be used to create and compare ObjectIDs. Mangrove offers serialization support for `bsoncxx::oid` so you can include them in your models as well. 
+
+## BSON Types
+
+The MongoDB C++ Driver offers a library called `libbsoncxx` that contains a set of utilities for using and manipulating BSON in C++. One of these utilities is a set of C++ types that cover every supported type in the {{% a_blank "BSON Specification" "http://bsonspec.org/spec.html" %}}.
+
+{{% notice warning %}}
+While the following types are all technically supported, we do not recommend using them if you can accomplish the same result with one of types discussed in the other sections. Many of the following types are dangerous and only used internally by MongoDB.
+{{% /notice %}}
+
+These types, which are all defined in {{% a_blank "`bsoncxx/types.hpp`" "http://mongodb.github.io/mongo-cxx-driver/types_8hpp_source.html" %}}, are all supported as serializable by Mangrove.
+
+The following types, which are in the `bsoncxx::types::` namespace are supported without any further configuration:
+
+* `b_double`
+* `b_bool`
+* `b_date`
+* `b_int32`
+* `b_int64`
+* `b_undefined`
+* `b_null`
+* `b_regex`
+* `b_timestamp`
+* `b_minkey`
+* `b_maxkey`
+* `b_oid`
+    - Not to be confused with the `bsoncxx::oid` discussed in the previous section.
+
+### BSON "View" Types
+
+Some of the types defined in the `bsoncxx::types::` namespace are non-owning "views" to their underlying data, meaning that they can be dangerous if the address they pointing to is freed from memory. The following types are BSON "views":
+
+* `b_utf8`
+* `b_document`
+* `b_array`
+* `b_binary`
+* `b_regex`
```

Comment:
`b_regex` appears in both of these lists, are the two lists supposed to be disjoint? 
My understanding is that you should inherit from `UnderlyingBSONDataBase` when using the types in the second list, and you don't need to when using types in the first list.